Address javadoc warnings with Java 21+

Found in https://github.com/strimzi/metrics-reporter/pull/73#issuecomment-2742840542. When running with Java 21 we get some javadoc warnings:
```
[WARNING] Javadoc Warnings
[WARNING] /Users/mickael/github/metrics-reporter/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/AbstractReporter.java:20: warning: use of default constructor, which does not provide a comment
[WARNING] public abstract class AbstractReporter {
[WARNING] ^
[WARNING] /Users/mickael/github/metrics-reporter/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/DataPointSnapshotBuilder.java:20: warning: use of default constructor, which does not provide a comment
[WARNING] public class DataPointSnapshotBuilder {
[WARNING] ^
[WARNING] /Users/mickael/github/metrics-reporter/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/HttpServers.java:20: warning: use of default constructor, which does not provide a comment
[WARNING] public class HttpServers {
[WARNING] ^
[WARNING] /Users/mickael/github/metrics-reporter/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/Listener.java:76: warning: use of default constructor, which does not provide a comment
[WARNING] public static class ListenerValidator implements ConfigDef.Validator {
[WARNING] ^
[WARNING] 4 warnings
```

--- a/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/AbstractReporter.java
+++ b/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/AbstractReporter.java
@@ -28,6 +28,11 @@ public abstract class AbstractReporter {
     private final Map<Object, MetricWrapper> disallowedMetrics = new ConcurrentHashMap<>();
 
     /**
+     * Empty constructor
+     */
+    public AbstractReporter() { }
+
+    /**
      * The current allowlist
      * @return A {@link Pattern} representing the allowlist
      */

--- a/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/DataPointSnapshotBuilder.java
+++ b/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/DataPointSnapshotBuilder.java
@@ -21,6 +21,10 @@ public class DataPointSnapshotBuilder {
 
     private static final Logger LOG = LoggerFactory.getLogger(DataPointSnapshotBuilder.class);
 
+    private DataPointSnapshotBuilder() {
+        // utility class that shouldn't be instantiated
+    }
+
     /**
      * Create a datapoint for a {@link InfoSnapshot} metric
      * @param labels The labels associated with the datapoint

--- a/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/HttpServers.java
+++ b/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/HttpServers.java
@@ -22,6 +22,10 @@ public class HttpServers {
     private final static Logger LOG = LoggerFactory.getLogger(HttpServers.class);
     private static final Map<Listener, ServerCounter> SERVERS = new HashMap<>();
 
+    private HttpServers() {
+        // utility class that shouldn't be instantiated
+    }
+
     /**
      * Get or create a new HTTP server if there isn't an existing instance for the specified listener.
      * @param listener The host and port

--- a/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/Listener.java
+++ b/client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/http/Listener.java
@@ -75,6 +75,11 @@ public class Listener {
      */
     public static class ListenerValidator implements ConfigDef.Validator {
 
+        /**
+         * Empty constructor
+         */
+        public ListenerValidator() { }
+
         @Override
         public void ensureValid(String name, Object value) {
             Matcher matcher = PATTERN.matcher(String.valueOf(value));

